connect/runtime/src/test/java/org/apache/kafka/connect/integration/RebalanceSourceConnectorsIntegrationTest.java

```diff
@@ -167,7 +171,11 @@ public void testReconfigConnector() throws Exception {
         // expect that we're going to restart the connector and its tasks
         StartAndStopLatch restartLatch = connectorHandle.expectedStarts(1);
 
-        // Reconfigure the source connector by changing the Kafka topic used as output
+        // expect that the connector will write the records to a new topic
+        connectorHandle.expectedRecords(anotherTopic, numRecordsProduced);
```

Comment:
Part 3a: Added the ability to expect the # of records the connector tasks write to (or read from) _specific topics_. This allows us to wait for the connector to write 100 records within a certain duration.